src/main/java/games/strategy/triplea/settings/SettingsWindow.java

```diff
@@ -1,195 +1,202 @@
+import java.awt.event.WindowEvent;
+import java.awt.event.WindowListener;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
+import javax.swing.BorderFactory;
 import javax.swing.Box;
+import javax.swing.JButton;
 import javax.swing.JComponent;
+import javax.swing.JDialog;
 import javax.swing.JLabel;
+import javax.swing.JOptionPane;
 import javax.swing.JPanel;
-import javax.swing.JScrollPane;
 import javax.swing.JTabbedPane;
 import javax.swing.JTextArea;
-import javax.swing.border.BevelBorder;
-
-import games.strategy.engine.ClientContext;
-import games.strategy.triplea.settings.ai.AiTab;
-import games.strategy.triplea.settings.battle.calc.BattleCalcTab;
-import games.strategy.triplea.settings.battle.options.BattleOptionsTab;
-import games.strategy.triplea.settings.folders.FoldersTab;
-import games.strategy.triplea.settings.scrolling.ScrollSettingsTab;
+import javax.swing.WindowConstants;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
 import games.strategy.ui.SwingComponents;
 
+
 /**
- * Window that contains a tabbed panel with preference categories, each tab contains fields that allow users to update
- * game settings. The window handles generic logic around preferences, each tab will specify configuration values for
- * the settings.
- *
- * <p>
- * Overall layout:
- * </p>
- *
- * <ul>
- * <li>Primary element is a JTabbed pain, the contents are organized into rows, one row per option presented to the
- * user.</li>
- * </ul>
- *
- * <p>
- * Each row consists of: label, swing input, detailed description
- * </p>
- *
- * <p>
- * Then we have some buttons:
- * </p>
- *
- * <ul>
- * <li>default (revert) settings</li>
- * <li>save settings</li>
- * <li>close window</li>
- * </ul>
+ * UI window with controls to update game settings and preferences, {@see ClientSetting}.
+ * Settings are grouped by type, the window consists of a TabbedPane and in it we load
+ * one tab per non-hidden {@code SettingType}.
+ * All data needed to render the settings UI is pulled from the {@code ClientSetting} enum.
  */
-public class SettingsWindow extends SwingComponents.ModalJDialog {
-
-  private static final long serialVersionUID = 8108714206041495198L;
-  private static final int MAX_WIDTH = 1100;
-  private static final int TEXT_LABEL_WIDTH = MAX_WIDTH / 4;
-  private static final int ROW_HEIGHT = 60;
-
-  /**
-   * Shows the settings window. The window is modal (a user therefore cannot open multiple at a time)
-   */
-  public static void showWindow() {
-    SwingComponents.showWindow(new SettingsWindow(
-        new BattleOptionsTab(ClientContext.battleOptionsSettings()),
-        new BattleCalcTab(ClientContext.battleCalcSettings()),
-        new AiTab(ClientContext.aiSettings()),
-        new ScrollSettingsTab(ClientContext.scrollSettings()),
-        new FoldersTab(ClientContext.folderSettings())));
+enum SettingsWindow {
```

Comment:
A few observations I noted when playing around with this UI:

1. Parent frame for the dialog is not set, so you get some undesirable UI artifacts like a) the default Java system icon, and b) a new item in the system tray.  Might have to add a `GameRunner#newJDialog()` method in order to capture the main frame like we've been doing recently.
1. The feature added in #1511 to display the setting default/range appears to no longer be present.  Was that intentional?  If so, should we not at least restore the legacy behavior to display the default/range, if applicable, in the description text area?

**EDIT:** Nevermind about (2).  I see that you changed the UI to display the valid range in a tooltip rather than directly in the panel.  The default is no longer displayed, but I guess that is less important.